crates/ide_assists/src/handlers/lambdify_for_each.rs

```diff
@@ -0,0 +1,388 @@
+    let method = ctx.find_node_at_offset::<ast::MethodCallExpr>()?;
+
+    let closure = match method.arg_list()?.args().next()? {
+        ast::Expr::ClosureExpr(expr) => expr,
+        _ => return None,
+    };
+
+    let (method, receiver) = validate_method_call_expr(ctx, method)?;
+
+    let param_list = closure.param_list()?;
+    let param = param_list.params().next()?.pat()?;
+    let body = closure.body()?;
+
+    let range = method.syntax().text_range();
+
+    let if_expr = match body.clone() {
+        Expr::IfExpr(if_expr) => {
+            if_expr
+        },
+        Expr::BlockExpr(block) => {
+            let mut stmts = block.statements();
+            let fst_stmt = stmts.next()?;
+            continue_iff(stmts.next().is_none())?; // Only one statement
+            // First statement is an expression...
+            let expr_stmt = match fst_stmt {
+                ast::Stmt::ExprStmt(expr_stmt) => expr_stmt,
+                _ => return None,
+            };
+
+            // ...and even an if clause...
+            let expr = expr_stmt.expr()?;
+            let if_expr = match expr {
+                ast::Expr::IfExpr(my_if_expr) => my_if_expr,
+                _ => return None,
+            };
+            if_expr
+        },
+        _ => return None,
+    };
+
+    let condition = if_expr.condition()?; // ... with a condition...
+    continue_iff(if_expr.else_branch().is_none()); // ... and no else branch...
+    let then_branch = if_expr.then_branch()?; // ... and a then branch
+
+    acc.add(
+        AssistId("convert_if_to_filter", AssistKind::RefactorRewrite),
+        "Replace this `if { ... }` with a `filter()`",
+        range,
+        |builder| {
+            let indent = method.indent_level();
+            
+            let mut buf = String::new();
+            // Recursively remove unnecessary `mut`s in the parameter
+            let pat_filter = param.clone_for_update();
+            let mut to_be_removed = vec![];
+            walk_pat(&pat_filter, &mut |cb|
+                if let Pat::IdentPat(ident) = cb {
+                    if let Some(mut_token) = ident.mut_token() {
+                        to_be_removed.push(mut_token);
+                    }
+                }
+            );
+            for mut_token in to_be_removed.into_iter() {
+                if let Some(ws) = mut_token.next_token().filter(|it| it.kind() == SyntaxKind::WHITESPACE) {
+                    ws.detach();
+                }
+                mut_token.detach();
+            }
+            format_to!(buf, "{}.filter(|&{}| {})", receiver, pat_filter, condition);
+
+            // Because we removed a if block, reident accordingly the rest of the block
+            let block = then_branch.clone_for_update();
+            block.reindent_to(indent);
+
+            format_to!(buf, ".for_each(|{}| {})", param, block);
+
+            builder.replace(range, buf)
+        },
+    )
+}
+
+// Assist: convert_sum_call
```

Comment:
I am unsure whether this assist has any value really, I don't see when someone would end up with a `for_each` exactly like this.